### PR TITLE
Move the position of checkmark in WS Invite Page

### PR DESCRIPTION
**Pull request by shubham1206agra**
### Fixed Issues

$ https://github.com/Expensify/App/issues/35717

### Tests

1. Open a Workspace where you want to add members.
2. Add new members using Invite Members flow.

- [x] Verify that no errors appear in the JS console

### Offline tests

Same as Tests

### QA Steps

Same as Tests

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos

<details>
<summary>Android: Native</summary>

</details>

<details>
<summary>Android: mWeb Chrome</summary>

Screenshot 2024-03-13 at 7 35 51 PM

</details>

<details>
<summary>iOS: Native</summary>

Screenshot 2024-03-13 at 7 25 11 PM

</details>

<details>
<summary>iOS: mWeb Safari</summary>

Screenshot 2024-03-13 at 7 32 14 PM

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

Screenshot 2024-03-13 at 5 18 35 PM

Screenshot 2024-03-13 at 5 19 37 PM

</details>

<details>
<summary>MacOS: Desktop</summary>

</details>

### CODE REVIEW

File: src/components/SelectionList/BaseListItem.tsx [hunk 1]
```diff
@@ -23,6 +24,7 @@ function BaseListItem<TItem extends ListItem>({
     onCheckboxPress,
     onDismissError = () => {},
     rightHandSideComponent,
+    checkmarkDirection = 'left',
```
Comment (mountiny) [inline src/components/SelectionList/BaseListItem.tsx:27]: Wouldn't position be better in this case?
```suggestion
    checkmarkPosition = 'left',
```

Comment (burczu) [inline src/components/SelectionList/BaseListItem.tsx:27]: This should not be done this way... we just need new ListItem type. Please check how we've done this with UserListItem, RadioListItem and TableListItem

**Review by burczu — CHANGES_REQUESTED** [review]

File: src/components/SelectionList/BaseListItem.tsx (hunk 1 above)
Comment (mountiny) [reply src/components/SelectionList/BaseListItem.tsx]: We will continue with this approach and make these changes later https://expensify.slack.com/archives/C01GTK53T8Q/p1710333711044009?thread_ts=1710332452.771549&cid=C01GTK53T8Q

**Review by mountiny — DISMISSED** [review]
Looks good to me, just waiting for the chekclist

**Comment by shawnborton** [comment]
Can you show me the "Before" screenshot too? I'm not quite sure what this fixed, so I want to make sure I am understanding that first.

**Comment by mountiny** [comment]
@shubham1206agra lets add web, ios native and android mweb

**Comment by shawnborton** [comment]
Ah okay, cool. I think that's fine and matches the design doc... but I have a feeling we're eventually going to make this view match the exact table style view we already have in Workspace settings? Either way, the adjustments look good to me.

**Comment by mountiny** [comment]
Thank you for checking!

@ishpaul777 can you please proceed with the checklist?

**Review by ishpaul777 — APPROVED** [review]
Looks good to me!

File: src/components/SelectionList/BaseListItem.tsx [hunk 2]
```diff
@@ -23,6 +24,7 @@ function BaseListItem<TItem extends ListItem>({
     onCheckboxPress,
     onDismissError = () => {},
     rightHandSideComponent,
+    checkmarkPosition = 'left',
```
Comment (luacmartins) [inline src/components/SelectionList/BaseListItem.tsx:27]: WE have consts for these
```suggestion
    checkmarkPosition = CONST.DIRECTION.LEFT,
```

File: src/components/SelectionList/BaseListItem.tsx [hunk 3]
```diff
@@ -76,7 +78,7 @@ function BaseListItem<TItem extends ListItem>({
                 style={pressableStyle}
             >
                 <View style={wrapperStyle}>
-                    {canSelectMultiple && (
+                    {canSelectMultiple && checkmarkPosition === 'left' && (
```
Comment (luacmartins) [inline src/components/SelectionList/BaseListItem.tsx:81]:
```suggestion
                    {canSelectMultiple && checkmarkPosition === CONST.DIRECTION.LEFT && (
```

File: src/components/SelectionList/BaseListItem.tsx [hunk 4]
```diff
@@ -99,6 +101,21 @@ function BaseListItem<TItem extends ListItem>({
 
                     {typeof children === 'function' ? children(hovered) : children}
 
+                    {canSelectMultiple && checkmarkPosition === 'right' && (
```
Comment (luacmartins) [inline src/components/SelectionList/BaseListItem.tsx:104]:
```suggestion
                    {canSelectMultiple && checkmarkPosition === CONST.DIRECTION.RIGHT && (
```

File: src/components/SelectionList/types.ts [hunk 5]
```diff
@@ -34,6 +34,9 @@ type CommonListItemProps<TItem> = {
     /** Component to display on the right side */
     rightHandSideComponent?: ((item: TItem) => ReactElement<TItem>) | ReactElement | null;
 
+    /** Direction of checkmark to show */
+    checkmarkPosition?: 'left' | 'right';
```
Comment (luacmartins) [inline src/components/SelectionList/types.ts:38]:
```suggestion
    checkmarkPosition?: CONST.DIRECTION.LEFT | CONST.DIRECTION.RIGHT;
```

File: src/components/SelectionList/types.ts [hunk 6]
```diff
@@ -267,6 +270,9 @@ type BaseSelectionListProps<TItem extends ListItem> = Partial<ChildrenProps> & {
     /** Component to display on the right side of each child */
     rightHandSideComponent?: ((item: ListItem) => ReactElement<ListItem>) | ReactElement | null;
 
+    /** Direction of checkmark to show */
+    checkmarkPosition?: 'left' | 'right';
```
Comment (luacmartins) [inline src/components/SelectionList/types.ts:274]:
```suggestion
    checkmarkPosition?: CONST.DIRECTION.LEFT | CONST.DIRECTION.RIGHT;
```

**Review by luacmartins — CHANGES_REQUESTED** [review]

**Review by luacmartins — APPROVED** [review]

**Pull request merged**